tasks/20230519-gauge-adder-v4/test/task.fork.ts

```diff
@@ -0,0 +1,144 @@
+    authorizer = await authorizerTask.deployedInstance('Authorizer');
+
+    // We'll need this task later on.
+    mainnetGaugeFactoryTask = new Task('20220822-mainnet-gauge-factory-v2', TaskMode.READ_ONLY, 'mainnet');
+  });
+
+  context('construction', () => {
+    it('stores the entrypoint', async () => {
+      expect(await gaugeAdder.getAuthorizerAdaptorEntrypoint()).to.equal(adaptorEntrypoint.address);
+    });
+
+    it('stores the gauge controller', async () => {
+      const gaugeControllerTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+      gaugeController = await gaugeControllerTask.deployedInstance('GaugeController');
+
+      // Ensure we can call functions on the gaugeController
+      const controllerAdmin = await gaugeController.admin();
+      expect(controllerAdmin).to.not.equal(ZERO_ADDRESS);
+      expect(await gaugeController.gauge_exists(ZERO_ADDRESS)).to.be.false;
+
+      expect(await gaugeAdder.getGaugeController()).to.equal(gaugeController.address);
+    });
+  });
+
+  context('advanced functions', () => {
+    let admin: SignerWithAddress;
+    let gauge: Contract;
+
+    before('load accounts', async () => {
+      admin = await getSigner(0);
+    });
+
+    before('create gauge factory', async () => {
+      const factoryTask = new Task('20220822-mainnet-gauge-factory-v2', TaskMode.TEST, getForkedNetwork(hre));
+      await factoryTask.run({ force: true });
+      factory = await factoryTask.deployedInstance('LiquidityGaugeFactory');
+
+      expect(await factory.isGaugeFromFactory(ZERO_ADDRESS)).to.be.false;
+    });
+
+    // We need to grant permission to the admin to add the LiquidityGaugeFactory to the GaugeAdder, and also to add
+    // gauges from said factory to the GaugeController.
+    before('grant permissions', async () => {
+      const addGaugeTypeAction = await actionId(gaugeAdder, 'addGaugeType');
+      const setFactoryAction = await actionId(gaugeAdder, 'setGaugeFactory');
+      const addGaugeAction = await actionId(gaugeAdder, 'addGauge');
+      const gaugeControllerAddGaugeAction = await actionId(
+        adaptorEntrypoint,
+        'add_gauge(address,int128)',
+        gaugeController.interface
+      );
+
+      await authorizer.connect(daoMultisig).grantRole(addGaugeTypeAction, admin.address);
+      await authorizer.connect(daoMultisig).grantRole(setFactoryAction, admin.address);
+      await authorizer.connect(daoMultisig).grantRole(addGaugeAction, admin.address);
+      await authorizer.connect(daoMultisig).grantRole(gaugeControllerAddGaugeAction, gaugeAdder.address);
+    });
+
+    it('can add a gauge type', async () => {
+      const tx = await gaugeAdder.connect(admin).addGaugeType('Ethereum');
+      const receipt = await tx.wait();
+
+      // `expectEvent` does not work with indexed strings, so we decode the pieces we are interested in manually.
```

Comment:
Thanks! Added this to the documentation "Testing Tips & Tricks," in case we run into this again.